SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -26,54 +27,72 @@ class ApplicationPresenter : ApplicationContract.Presenter() {
     override val coroutineContext: CoroutineContext
         get() = dispatchers.main + job
 
+
     override fun onViewTaken(view: ApplicationContract.View) {
         this.view = view
-        view.setStationNames(stations)
+        launch {
+           view.setStationNames(StationHelper.getStations())
+        }
     }
 
     private suspend fun getAPITimeData(
         originCrs: String,
         destinationCrs: String,
-        dateTime: String
+        dateTime: String,
+        adultCount: String,
+        childCount: String
     ): DepartureDetails {
         val url = URLBuilder("${baseURL}v1/fares?").apply {
             parameters["originStation"] = originCrs
             parameters["destinationStation"] = destinationCrs
             parameters["outboundDateTime"] = dateTime
-            parameters["numberOfChildren"] = "0"
-            parameters["numberOfAdults"] = "1"
+            parameters["numberOfChildren"] = childCount
+            parameters["numberOfAdults"] = adultCount
         }.build()
 
         return client.get {
             url(url)
         }
     }
 
-    override fun makeTrainSearch(originCrs: String, destinationCrs: String, dateTime: String) {
+    override fun makeTrainSearch(
+        originCrs: String,
+        destinationCrs: String,
+        dateTime: String,
+        adultCount: String,
+        childCount: String
+    ) {
         launch {
             try {
-                val departureDetails = getAPITimeData(originCrs, destinationCrs, dateTime)
+                val departureDetails =
+                    getAPITimeData(originCrs, destinationCrs, dateTime, adultCount, childCount)
 
                 val data = departureDetails.outboundJourneys.map {
```

Comment:
I would maybe put the initialisation of the DepartureInfo in a separate function for clarity